### DISCUSSION TITLE
Analysis across documents loaded into a single analyzer

### DISCUSSION THREAD

**Discussion by chklauser** [Eliah-Lakhin/lady-deirdre#13] [Q&A] [answered]
After reading [the book](https://lady-deirdre.lakhin.com/) and the examples in the repository, it's not clear to me, how I would implement an analysis that spans multiple documents. 

 * I see how the `AttrContext::read_doc` method lets me fetch a document given its ID, but the only doc ID I have inside a `Computable` impl is the doc ID I get from `context.node_ref().id` (the "current" document). I don't see a method to "enumerate" docs (other than trying out all possible IDs with `AttrContext::contains_doc`)
 * I could theoretically try to maintain my own index of documents outside the analyzer. But then I'm not clear on how to #12

**Comment by Eliah-Lakhin** [comment, marked as answer]
I hope that my answer in #12 addressed both questions. Regarding the document enumeration, I want to add that introducing this enumeration function is possible and not too difficult. I'm just not sure if this feature should be a part of the Analyzer's API. My vision is that it is essentially a part of the external state. However, if you have specific use cases where document enumeration is necessary, I will be happy to implement the corresponding function.

**Reply by chklauser** [reply]
My gut feeling is that you are right. The document alone is probably never going to be "enough". I'd always need additional information that puts the document into context. I _guess_ you could attach information to an `Attr` on the root node, but that seems hacky.
